components/service/glean/src/test/java/mozilla/components/service/glean/ping/MetricsPingTest.kt

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.ping
+
+import android.content.Context
+import org.junit.Test
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+class MetricsPingTest {
+    // NOTE: Using a "real" ApplicationContext here so that it will have
+    // a working SharedPreferences implementation
+    var applicationContext: Context = RuntimeEnvironment.application.applicationContext
+
+    @Test
+    fun `metrics ping has24HoursPassed() returns true if no value in shared prefs`() {
+        val metricsPing = MetricsPing(applicationContext)
+
+        // Clear the sharedPreferences on the MetricsPing so there is no last ping timestamp
+        metricsPing.metricsPreferences?.edit().clear().apply()
+
+        assertTrue(metricsPing.has24HoursPassed())
+    }
+
+    @Test
+    fun `metrics ping has24HoursPassed() returns false if 24 hours hasn't elapsed`() {
+        val metricsPing = MetricsPing(applicationContext)
+
+        // Clear the sharedPreferences on the MetricsPing so there is no last ping timestamp
+        metricsPing.metricsPreferences?.edit().clear().apply()
+
+        // Store a ping timestamp to compare to
+        metricsPing.updatePingTimestamp()
+
+        assertFalse(metricsPing.has24HoursPassed())
+    }
+
+    @Test
+    fun `metrics ping has24HoursPassed() returns true if 24 hours has elapsed`() {
+        val metricsPing = MetricsPing(applicationContext)
+
+        // Clear the sharedPreferences on the MetricsPing so there is no last ping timestamp
+        metricsPing.metricsPreferences?.edit().clear().apply()
+
+        // Store a ping timestamp from 25 hours ago to compare to
+        metricsPing.updatePingTimestamp(System.currentTimeMillis() - TimeUnit.HOURS.toMillis(25))
```

Comment:
The same consideration from the previous tests applies, we need to make sure that the expected value is being saved.

Reply:
I added some tests around storing of values to validate this.